### PR TITLE
Update to the shopping cart

### PR DESCRIPTION
**Pull request by Favourolotu**
Updated the readme partially.
Added option for removing from cart. 
Added choosing to add more than one book. 
Styling on cart page added. 
Recommendation service and page added.

### CODE REVIEW

File: src/main/java/com/example/bookstore/user/UserService.java [hunk 1]
```diff
@@ -22,19 +22,64 @@ public ShoppingCart getShoppingCart(String username) {
...
+
+            Set<Book> userPurchasedBooks = new HashSet<>(user.getPurchasedBooks()); // Get purchased books of the compared user
+            // Calculate Jaccard distance
+            Set<Book> union = new HashSet<>(currentUserPurchasedBooks);
+            union.addAll(userPurchasedBooks);
+            Set<Book> intersection = new HashSet<>(currentUserPurchasedBooks);
+            intersection.retainAll(userPurchasedBooks);
+            double jaccardDistance = (double) intersection.size() / (double) union.size();
+
+            if (jaccardDistance > 0.5) {
```
Comment (chase-scott) [inline src/main/java/com/example/bookstore/user/UserService.java:75]: Why did you choose 0.5?

**Review by chase-scott — APPROVED** [review]
Looks great!

File: src/main/java/com/example/bookstore/user/UserService.java (hunk 1 above)
Comment (Favourolotu) [reply src/main/java/com/example/bookstore/user/UserService.java:75]: I used it randomly, but if the purchases are similar by about 50% then we consider it.

File: src/main/java/com/example/bookstore/user/User.java [hunk 2]
```diff
@@ -30,6 +31,14 @@ public class User implements UserDetails {
     @JoinColumn(name = "cart_id", referencedColumnName = "id")
     private ShoppingCart shoppingCart;
 
+    @ManyToMany(cascade = CascadeType.ALL)
+    @JoinTable(
+            name = "user_purchased_books",
+            joinColumns = @JoinColumn(name = "user_id"),
+            inverseJoinColumns = @JoinColumn(name = "book_id"))
+    private Set<Book> purchasedBooks;
```
Comment (chase-scott) [inline src/main/java/com/example/bookstore/user/User.java:39]: Why is this a set?

**Pull request merged**